diff -Naur uboot-v2018.11-stm32mp-r3.1.org/include/configs/stm32mp1.h uboot-v2018.11-stm32mp-r3.1/include/configs/stm32mp1.h
--- uboot-v2018.11-stm32mp-r3.1.org/include/configs/stm32mp1.h	2019-11-07 15:33:31.807219806 +0100
+++ uboot-v2018.11-stm32mp-r3.1/include/configs/stm32mp1.h	2019-11-07 15:34:13.283270059 +0100
@@ -154,7 +154,6 @@
 	"if test ${boot_device} = serial || test ${boot_device} = usb;" \
 	"then stm32prog ${boot_device} ${boot_instance}; " \
 	"else " \
-		"run env_check;" \
 		"if test ${boot_device} = mmc;" \
 		"then env set boot_targets \"mmc${boot_instance}\"; fi;" \
 		"if test ${boot_device} = nand;" \
@@ -260,6 +259,7 @@
 	"altbootcmd=run bootcmd\0" \
 	"usb_pgood_delay=2000\0" \
 	"env_default=1\0"				\
+	"run env_check;" \
 	"env_check=if test $env_default -eq 1;"\
 		" then env set env_default 0;env save;fi\0" \
 	STM32MP_BOOTCMD \
